presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java

```diff
@@ -289,11 +313,21 @@ private static MethodDefinition generateProjectColumnarMethod(
 
         Variable positionCount = scope.declareVariable("positionCount", body, page.invoke("getPositionCount", int.class));
         Variable position = scope.declareVariable("position", body, constantInt(0));
-        Variable cardinality = scope.declareVariable("cardinality", body, selectedPositions.length());
         Variable outputBlock = scope.declareVariable(Block.class, "outputBlock");
         Variable blockBuilder = scope.declareVariable("blockBuilder", body, pageBuilder.invoke("getBlockBuilder", BlockBuilder.class, projectionIndex));
         Variable type = scope.declareVariable("type", body, pageBuilder.invoke("getType", Type.class, projectionIndex));
 
+        Variable cardinality;
+        BytecodeExpression loopCounter;
```

Comment:
I would rather call current for control variable `position` sth like `loopCounter` and use `position` as thing you are calling `loopCounter`.